graphql_compiler/tests/test_emit_output.py

```diff
@@ -29,7 +29,8 @@ def test_simple_immediate_output(self):
             MarkLocation(base_location),
             ConstructResult({
                 'foo_name': OutputContextField(base_name_location, GraphQLString),
-            })
+            }),
+            SelectWhereFilter({}),
```

Comment:
I don't think we want this block added if it isn't doing anything. Also, regardless of what the MATCH query looks like, it's strange from an IR perspective to construct results before filtering them again.

When this block appears, let's move it to right before `ConstructResult` instead of right after. That way, `ConstructResult` is always the last IR block, and that simplifies the error-checking code I commented on above.